tests/performance-tests/src/configuration/rampingRequestRateProfile.ts

```diff
@@ -0,0 +1,65 @@
+import { Options } from 'k6/options';
+import { mergeObjects, parseIntOptional } from '../utils/utils';
+
+interface Config {
+  // Name of the scenario.
+  scenario?: string;
+  // Total duration of the main stage of the test, providing a steady rate of requests.
+  mainStageDurationMinutes: number;
+  // The maximum requests per second to ramp up to during the main stage.
+  maxRequestRatePerSecond: number;
+  // Duration of a short final stage to allow any long-running requests to complete.
+  cooldownStageDurationMinutes: number;
+}
+
+const overrides: Partial<Config> = {
```

Comment:
I would suggest changing the types of `Config` to be optional if we expect them to be undefined at certain points rather than using the `Partial` operator. There's quite a few instances of `Partial` throughout this PR so this is the only comment I'm going to leave about it